File: ma-arving/src/main/java/ru/make/account/core/arving/web/dto/debt/DebtDto.java
package ru.make.account.core.arving.web.dto.debt;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DebtDto {
    private Long id;
    private Boolean isActive;
    private LocalDate createDate;
    private LocalDate closeDate;
    @NotNull(message = "Поле 'Название долга' должно быть заполнен")
    private String name;
    private String description;
    @NotNull(message = "Поле 'Сумма долга' должно быть заполнен")
    private BigDecimal sumDebt;
    private BigDecimal sumDebtCurrent;
    @NotNull(message = "Поле 'Счёт по долгу' должно быть заполнен")
    private Long accountId;
    @NotNull(message = "Поле 'Флаг должник' должно быть заполнен")
    private Boolean debtorFlag;
    private Boolean deleteFlag;
    private List<DebtOperationDto> debtOperations;
}

File: ma-portal/porta-arving/src/model/dto/DebtDto.ts
import Ticket from 'src/model/dto/TicketDto';
import {date} from 'quasar';

export interface DebtOperation {
  id?: number
  firstOperation: boolean
  ticket: Ticket
  debtorFlag: boolean
  sumOperation: number
}

export interface Debt {
  id?: number
  isActive: boolean
  createDate: string
  closeDate?: string
  name: string
  description: string
  sumDebt: number
  sumDebtCurrent: number
  accountId?: number
  debtorFlag: boolean
  deleteFlag: boolean
  debtOperations: DebtOperation[]
}

export function createEmptyDebt(): Debt {
  const nowDate = date.formatDate(Date.now(), 'YYYY-MM-DD');
  return {
    isActive: true,
    createDate: nowDate,
    name: '',
    description: '',
    sumDebt: 0,
    sumDebtCurrent: 0,
    debtorFlag: false,
    deleteFlag: false,
    debtOperations: [] as DebtOperation[]
  } as Debt
}

export const isValidName = (d: Debt) => {
  if (d) {
    return d.name && d.name.length > 0
  }
  return false;
}
export const isValidDate = (d: Debt) => {
  if (d) {
    return d.createDate && d.createDate.length > 0;
  }
  return false;
}
export const isValidSum = (d: Debt) => {
  if (d) {
    return Number(d.sumDebt) > 0;
  }
  return false;
}
export const isValidDebt = (d: Debt) => {
  return isValidName(d) && isValidDate(d) && isValidSum(d)
}
